src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -153,7 +159,24 @@ public function hydrate(object $component, array $data, string $componentName):
             } elseif (!$value && $type && $type->allowsNull() && is_a($type->getName(), \BackedEnum::class, true) && !\in_array($value, array_map(fn (\BackedEnum $e) => $e->value, $type->getName()::cases()))) {
                 $value = null;
             } elseif (null !== $value && $type && !$type->isBuiltin()) {
-                $value = $this->normalizer->denormalize($value, $type->getName(), 'json', [self::LIVE_CONTEXT => true]);
+                try {
+                    $value = $this->normalizer->denormalize($value, $type->getName(), 'json', [self::LIVE_CONTEXT => true]);
+                } catch(ExceptionInterface $exception) {
+                    $json = json_encode($value);
+                    $message = sprintf(
+                        'The normalizer was used to hydrate/denormalize the "%s" property on your "%s" live component, but it failed: %s',
+                        $name,
+                        get_class($component),
+                        $exception->getMessage()
+                    );
+
+                    // unless the data is gigantic, include it in the error to help
+                    if (strlen($json) < 1000) {
+                        $message .= sprintf(' The data sent from the frontend was: %s', $json);
+                    }
+
+                    throw new \LogicException($message, 0, $exception);
+                }
```

Comment:
An example of this:

<img width="829" alt="Screen Shot 2022-12-02 at 3 42 06 PM" src="https://user-images.githubusercontent.com/121003/205382161-4a1ee735-fecf-414e-b06d-e432838f3222.png">

I wish I could add line breaks, but at least all the info is there.